util/multi_concat.py

```diff
@@ -14,6 +14,50 @@
 from xarray.core.alignment import align
 
 
+try:
+    # solely for isinstance checks
+    import dask.array as dask_array
+
+    dask_array_type = (dask_array.Array,)
+except ImportError:  # pragma: no cover
+    dask_array_type = ()
+    dask_array = None
+
+
+def asarray(data):
+    return (
+        data
+        if (isinstance(data, dask_array_type) or hasattr(data, "__array_function__"))
+        else np.asarray(data)
+    )
+
+
+def lazy_array_equiv(arr1, arr2):
+    """Like array_equal, but doesn't actually compare values.
+       Returns True when arr1, arr2 identical or their dask names are equal.
+       Returns False when shapes are not equal.
+       Returns None when equality cannot determined: one or both of arr1, arr2 are numpy arrays;
+       or their dask names are not equal
+    """
+    if arr1 is arr2:
+        return True
+    arr1 = asarray(arr1)
+    arr2 = asarray(arr2)
+    if arr1.shape != arr2.shape:
+        return False
```

Comment:
Below, None is returned. Should that be returned here vs False?

Reply:
Per the function comments, True, False, and None are all valid return values from this function, each with a different meaning. None is used to indicate that equivalence could not be determined. This logic was pulled from a later version of xarray.